x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -141,181 +102,73 @@ const AssistantComponent: React.FC<Props> = ({
-  } = useFetchAnonymizationFields();
-
   const {
-    data: { data: allPrompts },
-    refetch: refetchPrompts,
-    isLoading: isLoadingPrompts,
-  } = useFetchPrompts();
-
-  const allSystemPrompts = useMemo(() => {
-    if (!isLoadingPrompts) {
-      return allPrompts.filter((p) => p.promptType === PromptTypeEnum.system);
-    }
-    return [];
-  }, [allPrompts, isLoadingPrompts]);
+    allPrompts,
+    allSystemPrompts,
+    anonymizationFields,
+    conversations,
+    isErrorAnonymizationFields,
+    isFetchedAnonymizationFields,
+    isFetchedCurrentUserConversations,
+    isFetchedPrompts,
+    isLoadingAnonymizationFields,
+    isLoadingCurrentUserConversations,
+    refetchPrompts,
+    refetchCurrentUserConversations,
+    setIsStreaming,
+  } = useDataStreamApis({ http, baseConversations, isAssistantEnabled });
 
   // Connector details
-  const { data: connectors, isFetchedAfterMount: areConnectorsFetched } = useLoadConnectors({
+  const { data: connectors, isFetchedAfterMount: isFetchedConnectors } = useLoadConnectors({
     http,
   });
   const defaultConnector = useMemo(() => getDefaultConnector(connectors), [connectors]);
 
-  const [currentConversationId, setCurrentConversationId] = useState<string | undefined>();
-
-  const [currentConversation, setCurrentConversation] = useState<Conversation | undefined>();
-
-  useEffect(() => {
-    if (setConversationTitle && currentConversation?.title) {
-      setConversationTitle(currentConversation?.title);
-    }
-  }, [currentConversation?.title, setConversationTitle]);
-
-  const refetchCurrentConversation = useCallback(
-    async ({
-      cId,
-      cTitle,
-      isStreamRefetch = false,
-    }: { cId?: string; cTitle?: string; isStreamRefetch?: boolean } = {}) => {
-      if (cId === '' || (cTitle && !conversations[cTitle])) {
-        return;
-      }
-
-      const conversationId = cId ?? (cTitle && conversations[cTitle].id) ?? currentConversation?.id;
-
-      if (conversationId) {
-        let updatedConversation = await getConversation(conversationId);
-        let retries = 0;
-        const maxRetries = 5;
-
-        // this retry is a workaround for the stream not YET being persisted to the stored conversation
-        while (
-          isStreamRefetch &&
-          updatedConversation &&
-          updatedConversation.messages[updatedConversation.messages.length - 1].role !==
-            'assistant' &&
-          retries < maxRetries
-        ) {
-          retries++;
-          await sleep(2000);
-          updatedConversation = await getConversation(conversationId);
-        }
-
-        if (updatedConversation) {
-          setCurrentConversation(updatedConversation);
-        }
-
-        return updatedConversation;
-      }
-    },
-    [conversations, currentConversation?.id, getConversation]
-  );
-
-  useEffect(() => {
-    if (areConnectorsFetched && conversationsLoaded && Object.keys(conversations).length > 0) {
```

Comment:
This `useEffect` has relocated to `use_current_conversation/index.tsx` and is slightly simplified and more complicated all at the same time